bot.md

```diff
@@ -0,0 +1,85 @@
+- Copy the output from the `scratch` script (ex: `/link ProgCrypto (Internal Test)`)
+- You will paste the output in the following step.
+
+### 4. PCD Join / Auth flow in Telegram
+
+- Make a new private group chat in the Telegram app (ex: `test chat`).
+- Go to the chat with your bot (ex: https://t.me/zulearn_bot)
+- Go to `Info`, then click `Add to group or channel`. Add the bot to `test chat`
+- Now, in `test chat`, type `/link`. You should get a message confirming that the test event was linked to the TG chat.
+- You can double check by using a GUI like Postico to view the `telegram_bot_events` table.
+
+### 5. Putting it All Together
+
+- In the Telegram app, go to the chat with your bot and type `/start`
+- Click `Generate ZKP`
+- Follow the link to PCDPass within Telegram (http://localhost:3000) and hit `Prove` when your ticket appears
+- If this is your first time, you will need to make a new PCDPass account:
+
+  - Make a new PCDPass account with an email you **know** has a ticket for the event in question (`ivan@0xparc.org` should work).
+  - _Note: This account will only exist locally on your device_
+
+- You should be returned to the Telegram app and presented with the `Send ZKP` option.
+- Click `Send ZKP`, then you will be redirected to `test_chat`
+
+### 6. (optional) Hot Reloading for TG Development
+
+This is a one-time setup for serving https locally. You need it for Telegram, because Telegram hates http and localhost. At the end, you will be able to access 127.0.0.1 (aka localhost) from a fake domain, `dev.local` (this is important for https)
+
+1. Install mkcert - [https://github.com/FiloSottile/mkcert](https://github.com/FiloSottile/mkcert)
+2. `mkcert -install`
+3. From root of repo: `mkdir apps/certificates && cd apps/certificates && mkcert dev.local`
+4. `sudo vi /etc/hosts` and add a line, `127.0.0.1 dev.local localhost`
+5. In `passport-client/.env` and `passport-server/.env`
+
+   ```python
+   # passport-client
+   IS_LOCAL_HTTPS=true
+   PASSPORT_SERVER_URL="https://dev.local:3002"
+
+   ```
+
+   ```python
+   # passport-server
+
+   # ... a bunch of other stuff
+   IS_LOCAL_HTTPS=true
+   PASSPORT_SERVER_URL="https://dev.local:3002"
+   PASSPORT_CLIENT_URL="https://dev.local:3000"
+   ```
+
+### Troubleshooting
+
+- When restarting your database, make sure to run `localdb:restart` not `yarn localdb:init && yarn localdb:up`.
+  - If you do accidentally run `init`, you will clear all local data. This is ok, but it causes some problems because your local PCDPass ticket still exists in localStorage in the Telegram Web View of PCDPass, but doesn't exist in the database.
+  - To rectify this situation, you should clear localStorage on Telegram PCDPass by right-clicking the on the Web view, then selecting `Inspect Element`. From there, you can go to the `Storage` tab and clear everything.
+  - Then, you just have to login again (email: dev@gmail.com, pw: devconnect)
```

Comment:
note: we should probably also just handle this better in passport-client. should be cognizant of the situation where localStorage pcds is out-of-date with the db state